app/api/v2/views/auth.py

```diff
@@ -0,0 +1,98 @@
+import json
+from flask import request, json
+from flask_restful import Resource
+from app.api.v2.models.users_model import UsersModel
+from app.api.v2.views.validation import ViewsValidation
+# from app.api.v2.auth_decorators.auth_decorator import isAdmin
+
+
+class AuthSignUp(Resource):
+
+    def post(self):
+
+        data = request.get_json(silent=True)
+
+        if not data:  # allow json formated data only
+
+            return ViewsValidation().views_error(
+                400, "Bad Request Format")
+
+        # check for missing fields
+        fields = ['username', 'email', 'password', 'phone_number']
+        if ViewsValidation().check_fields(fields, data):
+            # found missing fields
+            return ViewsValidation().check_fields(fields, data)
+
+        # instanciate Users model and pass request data
+        user = UsersModel(**data)
+
+        email = user.get_user_by_email(data["email"])
+
+        if email:
+            # duplicate user  error
+            return ViewsValidation().views_error(
+                202, "Duplicate User Error", "message")
+
+        # create user
+        response = user.create_user()
+
+        if response:
+            user_details = user.get_user_details(data["email"])
+            # generate token
+            auth_token = user.gen_auth_token(data["email"])
+
+            # user.decode_auth_token(auth_token)["role"]
+            # returns role or email
+
+            user_details.update({"token": auth_token.decode()})
+            return {  # user creation success return user data
+                "status": 201,
```

Comment:
avoid returning two status codes at the same time and use appropriate codes for each response.

Reply:
fix failing tests